tests/std/tests/P0896R4_views_filter_iterator/test.cpp

```diff
@@ -0,0 +1,187 @@
+
+        using C = typename iterator_traits<Iter>::iterator_category;
+        static_assert(is_same_v<typename I::iterator_category,
+            conditional_t<derived_from<C, bidirectional_iterator_tag>, bidirectional_iterator_tag,
+                conditional_t<derived_from<C, forward_iterator_tag>, forward_iterator_tag, input_iterator_tag>>>);
+
+        { // Validate iterator special member functions and base
+            static_assert(default_initializable<I> == default_initializable<Iter>);
+            if constexpr (default_initializable<Iter>) {
+                I defaultConstructed{};
+                assert(move(defaultConstructed).base().peek() == nullptr);
+                static_assert(is_nothrow_default_constructible_v<I>);
+            }
+
+            auto r0 = make_view();
+            I valueConstructed{r0, Iter{mutable_ints}};
+            static_assert(is_nothrow_constructible_v<I, R&, Iter>);
+
+            if constexpr (copyable<Iter>) {
+                I copyConstructed{valueConstructed};
+                assert(copyConstructed == valueConstructed);
+                static_assert(is_nothrow_copy_constructible_v<I>);
+
+                auto r1 = make_view();
+                I copyAssigned{r1, Iter{mutable_ints + 8}};
+                copyAssigned = copyConstructed;
+                assert(copyAssigned == valueConstructed);
+                static_assert(is_nothrow_copy_assignable_v<I>);
+                static_assert(same_as<const Iter&, decltype(as_const(copyConstructed).base())>);
+            }
+            assert(as_const(valueConstructed).base().peek() == mutable_ints);
+            assert(move(valueConstructed).base().peek() == mutable_ints);
+            static_assert(same_as<Iter, decltype(move(valueConstructed).base())>);
+        }
+
+        { // Validate sentinel constructors and base
+            S defaultConstructed{};
+            assert(defaultConstructed.base().peek() == nullptr);
+            static_assert(is_nothrow_default_constructible_v<S>);
+
+            auto r0 = make_view();
+            S valueConstructed{r0};
+            assert(valueConstructed.base().peek() == end(mutable_ints));
+
+            S copyConstructed{valueConstructed};
+            assert(copyConstructed.base().peek() == valueConstructed.base().peek());
+            static_assert(is_nothrow_copy_constructible_v<S>);
+
+            defaultConstructed = copyConstructed;
+            assert(defaultConstructed.base().peek() == valueConstructed.base().peek());
+            static_assert(is_nothrow_copy_assignable_v<S>);
+        }
+
+        { // Validate dereference ops
+            auto r0 = make_view();
+            auto i0 = r0.begin();
+            assert(*i0 == 0);
+            static_assert(noexcept(*i0));
+
+            assert(ranges::iter_move(i0) == 0); // NB: moving from int leaves it unchanged
+            // static_assert(noexcept(ranges::iter_move(i0)));

```

Comment:
Why is this commented out? No change requested.

Reply:
That is a very good question, will investigate